src/types/config.d.ts

```diff
@@ -1,9 +1,32 @@
 export type LxdConfigPair = Record<string, string>;
 
-export interface LxdConfigField {
-  default: string | boolean;
-  description: string;
+export type ConfigField = LxdConfigOption & {
+  category: string;
+  default: string;
   key: string;
-  scope: "global" | "local";
-  type: "string" | "integer" | "bool";
+};
+
+export interface LxdConfigOption {
+  defaultdesc?: string;
+  longdesc?: string;
+  scope?: "global" | "local";
+  shortdesc?: string;
+  type: "bool" | "string" | "integer";
+}
+
+export interface LxcConfigOptionCategories {
+  [category: string]: {
+    keys: {
+      [key: string]: LxdConfigOption;
+    }[];
+  };
+}
+
+export interface LxdConfigOptions {
+  configs: {
+    cluster: LxcConfigOptionCategories;
+    instance: LxcConfigOptionCategories;
+    project: LxcConfigOptionCategories;
+    server: LxcConfigOptionCategories;
```

Comment:
Related to Mason's question above - why are these 4 hard-coded here? Also, only `cluster` among them seems to be a "real" category that appears in the settings page. The other 3 are not there, and I don't see a bunch of others that are there instead (acme, core, images, etc).

Reply:
The response is structured in multiple hierarchies. Here on the top level we use only the part under server for the server settings. The part under instance holds configuration options for instances (like help texts and defaults).

The categories for the server settings are all under the server key here.

Reply:
It's very confusing, because they're at the top level when parsed in the settings form.

Reply:
Yeah it is a bit confusing from the server type here, the change is due to [this transition](https://github.com/canonical/lxd-ui/pull/543/files#diff-3ce5e2542470af50d3ea4bf0edb3ce1883fe5bb0b4300a5e90d808557cca83ecR65). I think the transition is good, because handle the transformed ConfigFields is way easier to handle in the UI, than dealing with the raw structure as returned from the API.